app/src/org/commcare/activities/EntitySelectActivity.java

```diff
@@ -210,21 +217,15 @@ public void onChanged() {
     }
 
     private void setSearchBannerState() {
-        if (!"".equals(adapter.getSearchQuery())) {
+        if (!"".equals(adapter.getSearchQuery()) || adapter.isFilteringByCalloutResult()) {

```

Comment:
Oh interesting. So does that mean that there has never been a way to clear your search on older devices?